packages/aws-cdk/lib/api/bootstrap/bootstrap-environment2.ts

```diff
@@ -0,0 +1,34 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import { BootstrapEnvironmentProps, deployStack, DeployStackResult, ISDK } from '..';
+
+export async function bootstrapEnvironment2(environment: cxapi.Environment, sdk: ISDK,
+                                            toolkitStackName: string, roleArn: string | undefined,
+                                            props: BootstrapEnvironmentProps = {}): Promise<DeployStackResult> {
+  const outdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk-bootstrap-new'));
+  const builder = new cxapi.CloudAssemblyBuilder(outdir);
+  const templateFile = `${toolkitStackName}.template.json`;
+
+  await fs.copy(
+    path.join(__dirname, 'bootstrap-template.json'),
+    path.join(builder.outdir, templateFile));
+
+  builder.addArtifact(toolkitStackName, {
+    type: cxapi.ArtifactType.AWS_CLOUDFORMATION_STACK,
+    environment: cxapi.EnvironmentUtils.format(environment.account, environment.region),
+    properties: {
+      templateFile,
+    },
+  });
+
+  const assembly = builder.buildAssembly();
+  return await deployStack({
+    stack: assembly.getStackByName(toolkitStackName),
+    sdk,
+    roleArn,
+    tags: props.tags,
+    execute: props.execute
+  });
```

Comment:
Perhaps extract this to `deployStackFromTemplate` as a more low-level building block and then just call it from here with the bootstrap template.

Also, I expected to see the bootstrap options passed in to CFN parameters...

Reply:
I tried extracting a `deployStackFromTemplate()` function, but it was super awkward (the file are created differently (`fs.writeJson()` versus `fs.copy()`, the temporary directories have different hints, we need to reference the `output` property of the `CloudAssemblyBuilder` to know where to put the files, etc.), so I abandoned it for now. If you have any suggestions how to make that part better, I'm all ears.

I've added passing the CLI options through the parameters.